nat-lab/tests/test_pq.py

```diff
@@ -130,3 +130,115 @@ async def test_pq_vpn_connection(
+                    ConnectionTag.DOCKER_CONE_CLIENT_1,
+                    stun_limits=ConnectionLimits(1, 1),
+                    nlx_1_limits=ConnectionLimits(2, 2),
+                ),
+                is_meshnet=False,
+            ),
+            "10.0.254.1",
+            marks=pytest.mark.linux_native,
+        ),
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.WINDOWS_VM_1,
+                adapter_type=AdapterType.WindowsNativeWg,
+                connection_tracker_config=generate_connection_tracker_config(
+                    ConnectionTag.WINDOWS_VM_1,
+                    stun_limits=ConnectionLimits(1, 1),
+                    nlx_1_limits=ConnectionLimits(2, 2),
+                ),
+                is_meshnet=False,
+            ),
+            "10.0.254.7",
+            marks=[
+                pytest.mark.windows,
+            ],
+        ),
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.WINDOWS_VM_1,
+                adapter_type=AdapterType.WireguardGo,
+                connection_tracker_config=generate_connection_tracker_config(
+                    ConnectionTag.WINDOWS_VM_1,
+                    stun_limits=ConnectionLimits(1, 1),
+                    nlx_1_limits=ConnectionLimits(2, 2),
+                ),
+                is_meshnet=False,
+            ),
+            "10.0.254.7",
+            marks=[
+                pytest.mark.windows,
+            ],
+        ),
+        # pytest.param(
+        #     SetupParameters(
+        #         connection_tag=ConnectionTag.MAC_VM,
+        #         adapter_type=AdapterType.BoringTun,
+        #         connection_tracker_config=generate_connection_tracker_config(
+        #             ConnectionTag.MAC_VM,
+        #             stun_limits=ConnectionLimits(1, 1),
+        #             nlx_1_limits=ConnectionLimits(2, 2),
+        #         ),
+        #         is_meshnet=False,
+        #     ),
+        #     "10.0.254.7",
+        #     marks=pytest.mark.mac,
+        # ),
+    ],
+)
+async def test_pq_vpn_rekey(
+    alpha_setup_params: SetupParameters,
+    public_ip: str,
+) -> None:
+    # Set rekey interval to some small value
+    alpha_setup_params.features.post_quantum_vpn.rekey_interval_s = 2
+
+    async with AsyncExitStack() as exit_stack:
+        env = await exit_stack.enter_async_context(
+            setup_environment(exit_stack, [alpha_setup_params])
+        )
+
+        client_conn, *_ = [conn.connection for conn in env.connections]
+        client_alpha, *_ = env.clients
+
+        ip = await testing.wait_long(stun.get(client_conn, config.STUN_SERVER))
+        assert ip == public_ip, f"wrong public IP before connecting to VPN {ip}"
+
+        await _connect_vpn_pq(
+            client_conn,
+            client_alpha,
+        )
+
+        await testing.wait_long(client_alpha.wait_for_log("Successful PQ REKEY"))
```

Comment:
I think the proper way to test it would be to make it actually fail after the server's timeout, so if the server should kill the session after like 5minutes, the testcase should wait for that amount of time at the very least to confirm it.

I see two testcases:
- Establish PQ and wait 5 minutes and check if we're still VPN connected( we shouldn't )
- Establish PQ and wait 5 minutes but at some time before we do a rekey and except the tunnel to be working after 5minutes

I think this is fine as that would show that functional correctness is achieved even though the testcase would be very long.

Maybe there's a way to configure the session length for the nordlynx?

Reply:
if going for such long test, don't forget to mark it as long `pytesst.mark.long` ... Although it will be run only during releases, otherwise it would be painful to run every pipeline :smile:

Reply:
So I've confirmed that the timeout is 5 min. I don't think it's feasible to wait that long but maybe there's a way to inspect `nlx` output for the changed key?

Reply:
I did it by inspecting the preshared key with nlx